lang/compiler.go

```diff
@@ -216,25 +215,19 @@ func (*TableObjectCompiler) CompilerType() flux.CompilerType {
 	panic("TableObjectCompiler is not associated with a CompilerType")
 }
 
-type DependenciesAwareProgram interface {
-	SetExecutorDependencies(execute.Dependencies)
+type LoggingProgram interface {
```

Comment:
Is there a way we can inject the logger into the context too?

Reply:
Yes, I think we can. The logger is actually needed to create the Executor and to make it log if something goes bad in the end. I inject the logger below (https://github.com/influxdata/flux/pull/1879/files#r325281074), in order to let a function create an Executor. The easiest way of doing this, should be to pass the logger to a Program upon Start.

Because actually, there is no need to inject it for making it available to the Program.

Reply:
I think we should defer this to a new PR